setup.py

```
@@ -0,0 +1,79 @@
+# Copyright (c) 2016-2018, The University of Texas at Austin
+# & University of California, Merced.
+#
+# All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the hIPPYlib library. For more information and source code
+# availability see https://hippylib.github.io.
+#
+# hIPPYlib is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License (as published by the Free
+# Software Foundation) version 2.0 dated June 1991.
+
+
+from setuptools import setup, find_packages
+from os import path
+from io import open
+
+
+try:
+    from dolfin import __version__ as dolfin_version
+except ImportError:
+    raise
+else:
+    if (int(dolfin_version[0:4]) < 2016):
```

We also need to raise an exception for FEniCS 2018